wallet/wallet.go

```diff
@@ -802,6 +803,7 @@ func (w *Wallet) recovery(chainClient chain.Interface,
 				)
 			})
 			if err != nil {
+				w.Manager.SetSyncedToMemory(&rollbackStamp)
```

Comment:
Hmm, so what was the initial error in the first place that got this out of sync? Sounds like all the `failed to fetch block hash for height 870852: block not found` errors are follow-up errors.
Would be nice to know that. But this fix definitely makes sense.

Reply:
I have the causing error in the issue for one instance.

Reply:
Ah, I see, fetching a block failed in a Neutrino setup.

Great catch, I'm quite certain this is indeed the cause for the errors you saw.

I wonder how we should deal with all the other cases where `SetSyncedTo()` is called within a database transaction that might be rolled back and cause the DB and in-memory state to de-synced...

Reply:
I think the out-of-synced state will make things worse. It looks like `recoverScopedAddresses` should be performed first before calling `SetSyncedTo`. Also it needs some refactor to move the non-db operations out so we don't need to wrap them in one giant tx `walletdb.Update`.